### DISCUSSION TITLE
Transaction Isolation for LMOVE Command in DragonflyDB

### DISCUSSION THREAD

**Discussion by suxb201** [dragonflydb/dragonfly#3979] [Q&A] [answered]
I am currently studying the DragonflyDB codebase and have learned a lot from it. However, I have encountered some confusion regarding the transaction logic, particularly with the LMOVE command.

From what I understand, the LMOVE command does not seem to guarantee transaction isolation. It appears to operate in two distinct phases: the first phase executes a PEEK operation, and the second phase executes PUSH and POP operations. If other commands modify the list between these two phases, could this potentially lead to data inconsistency?
<img width="1144" alt="image" src="https://github.com/user-attachments/assets/8210fbe5-1a2f-4dc0-8bf6-993f159deb5a">
<img width="1120" alt="image" src="https://github.com/user-attachments/assets/437fcaae-d1f4-4f11-b987-474c0c004e1e">

**Comment by romange** [comment, marked as answer]
As you can see in your editor, we execute the first hop with `conclude:false`. That means is that these keys stay logically "locked" for modifications until the transaction concludes. The second hop unlocks these keys.

**Reply by suxb201** [reply]
Thank you for your reply. ❤️
